### PR TITLE
feat(tower-plugin): create Endpoint for static ip in systemEndpoints

### PR DESCRIPTION
**Pull request by wu-cl**

### CODE REVIEW

File: plugin/tower/pkg/controller/endpoint/controller.go [hunk 1]
```diff
@@ -245,20 +298,137 @@ func (c *Controller) enqueueVMNics(vm *schema.VM) {
...
+	}
+}
+
+func (c *Controller) deleteEverouteCluster(old interface{}) {
+	if d, ok := old.(cache.DeletedFinalStateUnknown); ok {
+		old = d.Obj
+	}
+	cluster := old.(*schema.EverouteCluster)
+	for _, ctrl := range cluster.ControllerInstances {
+		if ctrl.IPAddr != "" {
```
Comment (KeyongSun) [inline plugin/tower/pkg/controller/endpoint/controller.go:342]: Who will response for the IP address validation?

File: plugin/tower/pkg/controller/endpoint/controller.go [hunk 2]
```diff
@@ -58,16 +60,33 @@ type Controller struct {
...
 const (
 	vnicIndex = "vnicIndex"
 	vmIndex   = "vmIndex"
 
-	ExternalIDName = "iface-id"
+	ExternalIDName         = "iface-id"
+	DefaultExternalIDValue = "default-externalID-value"
+
+	DynamicEndpointPrefix    = "tower.ep.dynamic"
+	StaticEndpointPrefix     = "tower.ep.static"
```
Comment (KeyongSun) [inline plugin/tower/pkg/controller/endpoint/controller.go:86]: add a '-' in the end?

Comment (wu-cl) [reply plugin/tower/pkg/controller/endpoint/controller.go:86]: These two mean TYPE, 	actually used are:
ControllerEndpointPrefix = "tower.ep.static.ctrl-"
SystemEndpointPrefix     = "tower.ep.static.sys-"

File: plugin/tower/pkg/controller/endpoint/controller.go (hunk 1 above)
Comment (wu-cl) [reply plugin/tower/pkg/controller/endpoint/controller.go:346]: No one will not valid it.
add IP valid on ADD and UPDATE op.

File: plugin/tower/pkg/controller/endpoint/controller.go (hunk 2 above)
Comment (KeyongSun) [reply plugin/tower/pkg/controller/endpoint/controller.go:86]: the "StaticEndpointPrefix" will be used in "getCtrlEndpointName", and the name will become as "tower.ep.static**xxxxx**-ip"

File: plugin/tower/pkg/controller/endpoint/controller.go [hunk 3]
```diff
@@ -382,6 +650,14 @@ func (c *Controller) getVMLabels(vmID string) (map[string]string, error) {
 	return labelsMap, nil
 }
 
+func (c *Controller) getCtrlEndpointName(cluster string, ctrl schema.EverouteControllerInstance) string {
+	return StaticEndpointPrefix + cluster + "-" + ctrl.IPAddr
```
Comment (KeyongSun) [inline plugin/tower/pkg/controller/endpoint/controller.go:654]: The  "StaticEndpointPrefix" is "ControllerEndpointPrefix"?

Comment (wu-cl) [reply plugin/tower/pkg/controller/endpoint/controller.go]: that's right, corrected!

**Review by KeyongSun — APPROVED** [review]

**Pull request merged**